experimental/codepointtrie/src/codepointtrie.rs

```diff
@@ -0,0 +1,906 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::error::{Error, FromDeserializedError};
+use crate::impl_const::*;
+use std::marker::PhantomData;
+use zerovec::ZeroVec;
+
+// Enums
+
+/// The width of the elements in the data array of a CodePointTrie.
+/// See UCPTrieValueWidth in ICU4C.
+#[derive(Clone, Copy, PartialEq)]
+pub enum ValueWidthEnum {
+    Bits16 = 0,
+    Bits32 = 1,
+    Bits8 = 2,
+}
+
+/// The type of trie represents whether the trie has an optimization that
+/// would make it small or fast.
+/// See UCPTrieType in ICU4C.
+#[derive(Clone, Copy, PartialEq)]
+pub enum TrieTypeEnum {
+    Fast = 0,
+    Small = 1,
+}
+
+// ValueWidth trait
+
+// AsULE is AsUnalignedLittleEndian, i.e. "allowed in a zerovec"
+pub trait ValueWidth: Copy + zerovec::ule::AsULE {
+    const ENUM_VALUE: ValueWidthEnum;
```

Comment:
Issue: I don't see where you use `ValueWidthEnum` or the associated type `ENUM_VALUE` on ValueWidth.

Reply:
It gets used by Auto